2200091/1949.py

```diff
@@ -0,0 +1,66 @@
+# 어디가 문제일까..........................
+
+import sys
+sys.stdin = open("input.txt")
+input = sys.stdin.readline
+
+def dfs(x, y, distance, is_construct):
+    global cnt
+
+    if cnt < distance:
+        cnt = distance
+
+    visited[x][y] = True
+
+    # 동서남북 확인
+    for dx, dy in delta:
+        nx = x + dx
+        ny = y + dy
+
+        # 범위를 벗어나거나 방문한 길이면 무시
+        if nx < 0 or nx >= N or ny < 0 or ny >= N or visited[nx][ny]:
+            continue
+
+        # 다음 길이 현재 길보다 낮으면 방문처리하고 그 자리에서 dfs
+        if graph[nx][ny] < graph[x][y]:
+            visited[nx][ny] = True
+            dfs(nx, ny, distance+1, is_construct)
+
```

Comment:
거의 다 풀어주셨네요! 👍 
```suggestion
            visited[nx][ny] = True
            dfs(nx, ny, distance + 1, is_construct)
            visited[nx][ny] = False
```
방문 후 다시 방문 안 한 상태로 돌려놓아야 합니다!